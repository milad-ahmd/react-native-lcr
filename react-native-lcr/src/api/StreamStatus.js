const StreamStatus = {
  Disconnected: 'disconnected',
  Connected: 'connected',
  Connecting: 'connecting',
};

export default StreamStatus;
