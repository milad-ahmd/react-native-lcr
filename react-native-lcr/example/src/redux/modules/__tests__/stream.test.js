import { actions, reducer, initialState } from "../stream"
import StreamStatus from "~/api/StreamStatus"
import snapshotTestModule from "./snapshotTestModule"
const { setStreamStatus, connectToStream } = actions

snapshotTestModule({
  name: "stream",
  reducer,
  actions: {
    setStreamStatus: setStreamStatus(StreamStatus.Connected),
    connectToStream: connectToStream(),
  },
  resetState: () => {
    return { ...initialState }
  },
})
