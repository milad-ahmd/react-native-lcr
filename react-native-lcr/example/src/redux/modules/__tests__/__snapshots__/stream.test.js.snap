// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators connectToStream matches snapshot 1`] = `
Object {
  "type": "STREAM/CONNECT_TO_STREAM",
}
`;

exports[`action creators setStreamStatus matches snapshot 1`] = `
Object {
  "payload": Object {
    "status": "connected",
  },
  "type": "STREAM/SET_STREAM_STATUS",
}
`;

exports[`stream module reducer #connectToStream matches snapshot 1`] = `undefined`;

exports[`stream module reducer #setStreamStatus matches snapshot 1`] = `
Array [
  DiffEdit {
    "kind": "E",
    "lhs": "disconnected",
    "path": Array [
      "status",
    ],
    "rhs": "connected",
  },
]
`;
